src/main/java/uk/gov/hmcts/juror/api/moj/repository/letter/court/ExcusalGrantedLetterListRepositoryImpl.java

```diff
@@ -1,79 +1,158 @@
+                .datePrinted(tuple.get(QJurorHistory.jurorHistory.dateCreated))
+                .isActive(tuple.get(QJurorPool.jurorPool.isActive))
+                .build())
+            .collect(Collectors.toList());
+
+        ConcurrentHashMap<String, ExcusalGrantedLetterList> excusalGrantedLetterListMap = new ConcurrentHashMap<>();
+        excusalGrantedLetterLists.forEach(excusalGrantedLetterList -> {
+            String jurorNumber = excusalGrantedLetterList.getJurorNumber();
+            if (excusalGrantedLetterListMap.containsKey(jurorNumber)) {
+                ExcusalGrantedLetterList existingExcusalGrantedLetterList =
+                    excusalGrantedLetterListMap.get(jurorNumber);
+                if (excusalGrantedLetterList.getDateExcused()
+                    .isAfter(existingExcusalGrantedLetterList.getDateExcused())) {
+                    excusalGrantedLetterListMap.put(jurorNumber, existingExcusalGrantedLetterList);
+                }
+            } else {
+                excusalGrantedLetterListMap.put(jurorNumber, excusalGrantedLetterList);
+            }
+        });
+
+        return excusalGrantedLetterListMap.values().stream().toList();
     }
 
+    private JPAQuery<Tuple> buildBaseQuery() {
+
+        JPAQueryFactory queryFactory = new JPAQueryFactory(entityManager);
+
+        return queryFactory.select(
+                QJurorPool.jurorPool.pool.poolNumber,
+                QJuror.juror.jurorNumber,
+                QJuror.juror.firstName,
+                QJuror.juror.lastName,
+                QJuror.juror.postcode,
+                QJurorStatus.jurorStatus.statusDesc,
+                QJuror.juror.excusalDate,
+                QJurorHistory.jurorHistory.dateCreated,
+                QJurorPool.jurorPool.isActive,
+                EXCUSAL_CODE.excusalCode.description)
+            .from(QJurorPool.jurorPool)
+            .join(QJuror.juror).on(QJuror.juror.eq(QJurorPool.jurorPool.juror))
+            .join(QPoolRequest.poolRequest)
+            .on(QPoolRequest.poolRequest.eq(QJurorPool.jurorPool.pool))
+            .join(QJurorStatus.jurorStatus).on(QJurorStatus.jurorStatus.eq(QJurorPool.jurorPool.status))
+            .join(EXCUSAL_CODE.excusalCode).on(EXCUSAL_CODE.excusalCode.code.eq(QJuror.juror.excusalCode))
+            .leftJoin(QJurorHistory.jurorHistory)
+            .on((QJurorHistory.jurorHistory.jurorNumber.eq(QJuror.juror.jurorNumber))
+                .and(QJurorHistory.jurorHistory.poolNumber.eq(QPoolRequest.poolRequest.poolNumber))
+                .and(QJurorHistory.jurorHistory.historyCode.eq(HistoryCodeMod.EXCUSED_LETTER))
+                .and(QJurorHistory.jurorHistory.dateCreatedDateOnly
+                    .after(QJuror.juror.bureauTransferDate)))
+            .where(QJurorPool.jurorPool.status.status.eq(IJurorStatus.EXCUSED)
+                .and(QPoolRequest.poolRequest.owner.eq(SecurityUtil.getActiveOwner())));
```

Comment:
Should be QJurorPool.pool.owner filter instead of pool

Reply:
updated